### PR TITLE
[now-client] Use `@zeit/fetch` instead of `node-fetch`

### PR DESCRIPTION
**Pull request by lucleray**
We want to take advantage of the better defaults and retry policies (among others).

This bumps `@zeit/fetch` and replaces `node-fetch` with `@zeit/fetch` in `now-client`.

Also update retry policy for "uploading" files to:
- base 10ms
- 5 retries
- factor 6

which means the timeouts are 10, 60, 360, 2160, 12960

### CODE REVIEW

**Review by leo — APPROVED** [review]
The code looks great!

File: yarn.lock [hunk 1]
```diff
@@ -9731,7 +9738,7 @@ shellwords@^0.1.1:
   resolved "https://registry.yarnpkg.com/shellwords/-/shellwords-0.1.1.tgz#d6b9181c1a48d397324c84871efbcfc73fc0654b"
   integrity sha512-vFwSUfQvqybiICwZY5+DAWIPLKsWO31Q91JSKl3UYv+K5c2QRPzn0qzec6QPu1Qc9eHYItiP3NdJqNVqetYAww==
 
-signal-exit@3.0.2, signal-exit@TooTallNate/signal-exit#update/sighub-to-sigint-on-windows, signal-exit@^3.0.0, signal-exit@^3.0.2:
```
Comment (styfle) [inline yarn.lock:9734]: @lucleray @TooTallNate We need to fix this. Perhaps we should publish a fork of signal-exit to avoid this issue?

Comment (lucleray) [reply yarn.lock:9734]: @styfle https://github.com/zeit/now/pull/3508

Comment (lucleray) [reply yarn.lock:9734]: Thanks for reporting 🙂

**Pull request merged**